pkg/rancher-components/src/components/StringList/StringList.vue

```diff
@@ -0,0 +1,572 @@
+      }
+    },
+
+    /**
+     * Show/Hide the in-line editing to edit an existing item
+     */
+    toggleEditMode(show: boolean, item?: string) {
+      if (show) {
+        this.toggleCreateMode(false);
+        this.value = this.isEditItem;
+
+        this.isEditItem = item || '';
+        this.setFocus(INPUT.edit);
+      } else {
+        this.value = null;
+        this.toggleError('duplicate', false);
+        this.onSelectLeave();
+
+        this.isEditItem = null;
+      }
+    },
+
+    getRef(id: string) {
+      const ref = this.$refs[id];
+
+      return (Array.isArray(ref) ? ref[0] : ref) as HTMLElement;
+    },
+
+    /**
+     * Create a new item and insert in the items list
+     */
+    saveItem() {
+      if (this.value) {
+        const items = [
+          ...this.items,
+          this.value.trim(),
+        ] as string[];
+
+        if (findStringDuplicates(items, this.caseSensitive)) {
+          this.toggleError('duplicate', true, INPUT.create);
+
+          return;
+        }
+
+        this.updateItems(items);
+      }
+      this.toggleCreateMode(false);
+    },
+
+    /**
+     * Update an existing item in the items list
+     */
+    updateItem(item: string) {
+      if (this.value) {
+        const items = [...this.items] as string[];
+        const index = findStringIndex(items, item, false);
+
+        if (index !== -1) {
+          items[index] = this.value.trim();
+        }
+
+        if (findStringDuplicates(items, this.caseSensitive)) {
+          this.toggleError('duplicate', true, INPUT.edit);
+
+          return;
+        }
+
+        this.updateItems(items);
+      }
+      this.toggleEditMode(false);
+    },
+
+    /**
+     * Delete item and update items list
+     */
+    deleteItem(item?: string): number {
+      let index = -1;
+
+      if (item) {
+        index = findStringIndex(this.items as string[], item, false);
+        const items = this.items.filter(f => f !== item) as string[];
+
+        this.updateItems(items);
+      }
+
+      return index;
+    },
+
+    /**
+     * Update items list and emit to parent component
+     */
+    updateItems(items: string[]) {
+      this.$emit('change', items);
+    },
+  },
+
+});
+</script>
+
+<template>
+  <div
+    class="string-list"
+    :class="{ readonly }"
+  >
+    <div
+      ref="box"
+      data-testid="div-string-list-box"
+      class="string-list-box"
+      tabindex="0"
+      @dblclick="onClickEmptyBody()"
+    >
+      <div
+        v-for="item in items"
+        :key="item"
+        :ref="item"
+        :class="{
+          selected: selected === item,
+          readonly
+        }"
+        :data-testid="`div-item-${item}`"
+        class="item static"
+        tabindex="0"
+        @mousedown="onSelect(item)"
+        @dblclick.stop="toggleEditMode(true, item)"
+        @keydown.down.prevent="onSelectNext('down')"
+        @keydown.up.prevent="onSelectNext('up')"
+        @blur="onSelectLeave(item)"
```

Comment:
I just realized that the input closes in different ways between creation and modification. It may be worth to keep it consistent and allow to close it in the same way.